src/physics/burgers.cpp

```diff
@@ -142,9 +156,26 @@ template <int dim, int nstate, typename real>
 std::array<real,nstate> Burgers<dim,nstate,real>
 ::source_term (
     const dealii::Point<dim,real> &pos,
-    const std::array<real,nstate> &/*solution*/) const
+    const std::array<real,nstate> &/*solution*/,
+    const real current_time) const
 {
     std::array<real,nstate> source;
+
+    using TestType = Parameters::AllParameters::TestType;
+
+    if(this->test_type == TestType::burgers_energy_stability){
```

Comment:
Instead of passing a test type, you could derive a class from `Burgers` called `BurgersEnergyStability` to `override` the `source_term` function, similar to what @donovan97 had done with the `BurgersRewienski` class, and the `PhysicsFactory` would handle choice of which class to use. Then you would add a PDE Type related to this new class and use that in your `.prm` files.

Reply:
But this is a lot of work to simply change the `source_term()`, maybe there should be a `SourceTermFactory` or something? @donovan97 @dougshidong

Reply:
This is only a temporary fix because in the future manufactured solution should be able to handle time dependence. Moving forward, after this pull request, I think the best approach would be to put time dependence in the manufactured solution class, then none of the source terms will need to know the test types.

Reply:
fixed as per issue #126

Reply:
Resolving as it will be fixed in the above issue